### PR TITLE
refactor: `SelectionSort` like classes and their tests

### PR DESCRIPTION
**Pull request by alxkm**
Refactor: Adding test common approach, adding javadocs, renaming variables

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

**Comment by alxkm** [comment]
In this pull request, also changed SortingAlgorithmTest to customize size of generated array. For recursive selection sort, time to time this was an stackOverFlow, to avoid  failed builds, size of generated arrray can be changed by demand.

**Comment by alxkm** [comment]
Hello @vil02 

This PR should fix this problem https://github.com/TheAlgorithms/Java/actions/runs/9708998343/job/26796827946

I already runned this bunch of times and no such cases was.

File: src/main/java/com/thealgorithms/sorts/SelectionSort.java [hunk 1]
```diff
@@ -1,46 +1,38 @@
...
+        for (int i = 0; i < array.length - 1; i++) {
+            // Find the minimum element in the unsorted array
             int minIndex = i;
-            for (int j = i + 1; j < n; j++) {
-                if (arr[minIndex].compareTo(arr[j]) > 0) {
+            for (int j = i + 1; j < array.length; j++) {
+                if (array[j].compareTo(array[minIndex]) < 0) {
                     minIndex = j;
                 }
             }
```
Comment (vil02) [inline src/main/java/com/thealgorithms/sorts/SelectionSort.java:29]: This deserves its own method.

File: src/main/java/com/thealgorithms/sorts/SelectionSort.java [hunk 2]
```diff
@@ -1,46 +1,38 @@
...
      */
     @Override
-    public <T extends Comparable<T>> T[] sort(T[] arr) {
-        int n = arr.length;
-        for (int i = 0; i < n - 1; i++) {
+    public <T extends Comparable<T>> T[] sort(T[] array) {
+        if (array.length == 0) {
+            return array;
+        }
+
```
Comment (vil02) [inline src/main/java/com/thealgorithms/sorts/SelectionSort.java:20]:
```suggestion
```

File: src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java
Comment (vil02) [inline src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java:1]: `recursiveSelectionSort` could be simplified to:

```java
    private static <T extends Comparable<T>> void recursiveSelectionSort(T[] array, int index) {
        if (index == array.length - 1) {
            return;
        }

        SortUtils.swap(array, index, findMinIndex(array, index));

        // Recursively call selection sort for the remaining array
        recursiveSelectionSort(array, index + 1);
    }
```

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java
Comment (alxkm) [reply src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java:1]: Thanks. Now this looks very clear

File: src/main/java/com/thealgorithms/sorts/SelectionSort.java (hunk 2 above)
Comment (alxkm) [reply src/main/java/com/thealgorithms/sorts/SelectionSort.java]: This is already removed according to this comment https://github.com/TheAlgorithms/Java/pull/5265#discussion_r1659318388

File: src/main/java/com/thealgorithms/sorts/SelectionSort.java (hunk 1 above)
Comment (alxkm) [reply src/main/java/com/thealgorithms/sorts/SelectionSort.java]: This also already made in this suggestion https://github.com/TheAlgorithms/Java/pull/5265#discussion_r1659318388

File: src/main/java/com/thealgorithms/sorts/SelectionSort.java [hunk 3]
```diff
@@ -1,46 +1,38 @@
...
-        }
-
-        String[] strings = {"c", "a", "e", "b", "d"};
-        String[] sortedStrings = selectionSort.sort(strings);
-        for (int i = 0; i < sortedStrings.length - 1; ++i) {
-            assert strings[i].compareTo(strings[i + 1]) <= 0;
-        }
+        return array;
     }
 }
```
[the remark this answers is not in the document]
Comment (alxkm) [reply src/main/java/com/thealgorithms/sorts/SelectionSort.java]: Now it is transparent, thanks

**Review by vil02 — APPROVED** [review]

**Pull request merged**